private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/maven/MavenResolver.java

```diff
@@ -368,19 +368,37 @@ private List<Dependency> createManagedDependencies(
     // Without this, the versions requested in BOMs will be preferred to the
     // one that the user requested
     // First add all the dependencies that have versions
+    Map<String, String> artifactVersions = new HashMap<>();
     Set<String> artifactKeys = new HashSet<>();
     dependencies.stream()
         .map(Dependency::getArtifact)
         .filter(artifact -> artifact.getVersion() != null)
         .filter(artifact -> !artifact.getVersion().isEmpty())
-        .forEach(artifact -> artifactKeys.add(getArtifactKey(artifact)));
+        .forEach(
+            artifact -> {
+              artifactVersions.put(getArtifactKey(artifact), artifact.getVersion());
+              artifactKeys.add(getArtifactKey(artifact));
+            });
 
     ImmutableList.Builder<Dependency> managedDependencies = ImmutableList.builder();
     managedDependencies.addAll(dependencies);
 
     // Then add all the BOM dependencies that are not in the declared dependency list
+    boolean rjeVerbose = System.getenv("RJE_VERBOSE") != null;
     bomDependencies.stream()
-        .filter(dep -> !artifactKeys.contains(getArtifactKey(dep.getArtifact())))
+        .filter(
+            dep -> {
+              String artifactKey = getArtifactKey(dep.getArtifact());
```

Comment:
Nit: This could be done in a `peek`, but I guess here is fine too

Reply:
I think you're right, the peek is better.  Added in https://github.com/bazelbuild/rules_jvm_external/pull/1178